pkg/actuators/machine/utils.go

```diff
@@ -132,6 +132,56 @@ func getInstanceByID(id string, client awsclient.Client, instanceStateFilter []*
 	return instance, instanceHasAllowedState(instance, instanceStateFilter)
 }
 
+// correctExistingTags validates Name and clusterID tags are correct on the instance
+// and sets them if they are not.
+func correctExistingTags(machine *machinev1.Machine, instance *ec2.Instance, client awsclient.Client) error {
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.CreateTags
+	if instance == nil || instance.InstanceId == nil {
+		return fmt.Errorf("unexpected nil found in instance: %v", instance)
+	}
+	clusterID, ok := getClusterID(machine)
+	if !ok {
+		return fmt.Errorf("unable to get cluster ID for machine: %q", machine.Name)
+	}
+	nameTagOk := false
+	clusterTagOk := false
+	for _, tag := range instance.Tags {
+		if tag.Key != nil && tag.Value != nil {
+			if *tag.Key == "Name" && *tag.Value == machine.Name {
+				nameTagOk = true
+			}
+			if *tag.Key == "kubernetes.io/cluster/"+clusterID && *tag.Value == "owned" {
+				clusterTagOk = true
+			}
+		}
+	}
+
+	// Update our tags if they're not set or correct
+	if !(nameTagOk && clusterTagOk) {
```

Comment:
Nit: I think technically this would be better as you're more likely to short circuit early this way around (we expect nameTagOk to be true most of the time right?)
```suggestion
	if !nameTagOk || !clusterTagOk {
```

Reply:
Yeah, I think you're right about this, but the this is definitely in the realm of the compiler optimizing things I would think.

Reply:
Haha, I would hope so too, but we can help the compiler out 😅